### PR TITLE
Updated lack of HSTS header finding

### PR DESCRIPTION
**Pull request by RRudder**
* Significant update to template.md 
* addition of recommendations.md

### CODE REVIEW

File: submissions/description/server_security_misconfiguration/lack_of_security_headers/strict_transport_security/template.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Lack of HTTP Strict Transport Security Header
+
+## Overview of the Vulnerability
+
+A lack of HTTP response headers can lead to sensitive data being transmitted unencrypted and being intercepted while in transit.There are multiple HTTP response headers used in communication between the server and client which can be implemented to improve security against well documented vulnerabilities. For example, the HTTP Strict Transport Security (HSTS) header instructs the website to enforce the use of HTTPS.
+
+By not having a HSTS header in place, the application fails to enforce users to connect to it over an encrypted connection. If an attacker is positioned to intercept or modify a user’s network traffic, they can perform a protocol downgrade attack, a Machine-in-the-Middle (MitM) attack, or hijack a user’s cookies. This aids an attacker in gaining access to the data being sent between the server and client, such as login credentials or application cookies.
```
Comment (evildaemond) [inline submissions/description/server_security_misconfiguration/lack_of_security_headers/strict_transport_security/template.md:7]: Mild direct improvements for this templates overview

A lack of the HTTP response header for Strict Transport Security can lead to sensitive data being transmitted unencrypted and being intercepted while in transit. There are multiple HTTP response headers used in communication between the server and client which can be implemented to improve security against well documented vulnerabilities. For example, the HTTP Strict Transport Security (HSTS) header instructs the website to enforce the use of HTTPS.

By not having a HSTS header in place, the application fails to enforce users to connect to it over an encrypted connection. If an attacker is positioned to intercept or modify a user’s network traffic, they can perform a protocol downgrade attack, a Persons-in-the-Middle (PitM) attack, or hijack a user’s cookies. This aids an attacker in gaining access to the data being sent between the server and client, including the login credentials or application cookies.

File: submissions/description/server_security_misconfiguration/lack_of_security_headers/strict_transport_security/template.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+## Business Impact
+
+Not having a HSTS header can lead to reputational damage and indirect financial loss to the business due to an attacker’s access to data over an unencrypted HTTP connection. The degree of impact is dependent on the sensitivity of data being transmitted over the wire.
+
+## Steps to Reproduce
+
+1. Enable a HTTP interception proxy, such as Burp Suite or OWASP ZAP
+1. Navigate to the following endpoint using a browser: {{URL}}
+1. Capture the request using the HTTP interception proxy
+1. Observe that a HSTS header is not used within the HTTP headers
```
Comment (evildaemond) [inline submissions/description/server_security_misconfiguration/lack_of_security_headers/strict_transport_security/template.md:18]: A direct technical update here, we don't review the request we review the response

1. Capture the request using a HTTP intercept proxy
1. Observe that a HSTS header is not in the response of the HTTP request

**Review by evildaemond — CHANGES_REQUESTED** [review]
2 direct changes to be made, then LGTM

File: submissions/description/server_security_misconfiguration/lack_of_security_headers/strict_transport_security/template.md (hunk 1 above)
Comment (RRudder) [reply submissions/description/server_security_misconfiguration/lack_of_security_headers/strict_transport_security/template.md]: Cheers for the feedback, please see latest commit

**Review by evildaemond — APPROVED** [review]
LGTM

**Pull request merged**